packages/core/lib/util/get-response.ts

```diff
@@ -0,0 +1,139 @@
+import { ExecutionResult } from "graphql";
+import { ExecutionPatchResult } from "../types";
+import { calculateByteLength } from "./calculate-byte-length";
+import { ReadableStream, Response } from "cross-undici-fetch";
+
+export type TransformResultFn = (result: ExecutionResult | ExecutionPatchResult) => any;
+export const DEFAULT_TRANSFORM_RESULT_FN: TransformResultFn = (result: ExecutionResult) => result;
+
+export function getRegularResponse(executionResult: ExecutionResult, transformResult = DEFAULT_TRANSFORM_RESULT_FN): Response {
+  const transformedResult = transformResult(executionResult);
+  const responseBody = JSON.stringify(transformedResult);
+  const contentLength = calculateByteLength(responseBody);
+  const headersInit: HeadersInit = {
+    "Content-Type": 'application/json',
+    "Content-Length": contentLength.toString()
+  };
+  const responseInit: ResponseInit = {
+    headers: headersInit,
+    status: 200,
+  };
+  return new Response(responseBody, responseInit);
+}
+
+export function getMultipartResponse(
+  asyncExecutionResult: AsyncIterable<ExecutionPatchResult<any>>,
+  transformResult = DEFAULT_TRANSFORM_RESULT_FN
+): Response {
+  const headersInit: HeadersInit = {
+    Connection: "keep-alive",
+    "Content-Type": 'multipart/mixed; boundary="-"',
+    "Transfer-Encoding": "chunked",
+  };
+  const responseInit: ResponseInit = {
```

Comment:
responseInit should be exposed through an argument, as a user may want to flush extra headers. eg https://github.com/maraisr/piecemeal/blob/2d9121904c33d0d0e6f606752f7250d590e2991b/src/worker.ts#L16-L17

Reply:
User can do it with `response.headers.append` method.

Reply:
Headers, but what about other properties? And fairly sure once a response is created its immutable, or CF throws a hissy fit.

Reply:
You can copy the parts of the Response object and create your own. Adding this kind of configurations will make this function overcomplicated so I'll keep it in that way.